### ISSUE TITLE
Compiler logs should use consistent addressing formatting

### ISSUE THREAD

**Issue by 0xdaryl** [eclipse/omr#5991]
In any typical compiler trace log there is no consistency in the formatting of addresses.  For instance, addresses are formatted:

* with/without leading zeros
* with/without  a`0x` prefix.  I recall in the past some addresses even had a double prefix (`0x0x`) though I haven't seen one recently.
* upper/lower case
* literally every combination of the above
* on x86, some addresses are prefixed with a `0` and suffixed with an `h` in the MASM syntax style

In addition to being aesthetically displeasing it can make searching on a particular address difficult because of the changes in letter case.  Granted, some formatting styles have been chosen for the context in which they appear in the log (e.g., where spacing and alignment are being carefully controlled), but most addresses have no such requirement. 

I propose we choose a single style for printing addresses and make it consistent.  My personal preference is `0xaabbccdd` without leading zeros.  Having the prefix makes it clear this is a hex value and it is easy to swipe and paste the address into a debugger for immediate use without having to manually prefix it with `0x` first  The appropriate OMR format specifier macro in `include_core/omrformatconsts.h` should be used.

**Comment by Leonardo2718** [comment]
Somewhat related to this: a debugging feature that I've found useful a few times in the past is the `enumerateAddresses` option. It essentially replaces (many) addresses in log files with names that are stable across multiple runs of the same program. So, in theory, the same object allocated in two different runs will have the same "address name" even if the memory addresses it gets allocated at are different. This is useful, for example, in combination with `breakOnCreate{}` in order to figure out when, where, and why a particular object gets allocated.

Unfortunately, as new trace messages have been added over time, there are a few places that unconditionally print the memory address instead of the address name (I'm likely guilty of having introduced a few of those myself), which limits the usability of this feature.

It would be good to find all the places that should be printing the address names, but aren't currently, and fix them. (I didn't keep track of the ones I saw, unfortunately)

I should note that I don't think this feature is necessary for _all_ addresses in a log but should be available when printing important data structures like nodes, treetops, instructions, etc.

**Comment by 0xdaryl** [comment]
Yes, that's a good suggestion @Leonardo2718.  I suspect some of the reasons those aren't used as much as they should be are lack of awareness and the fact that it is easier to just dump an address than to first find the debug object and then dispatch a function out of it.  Overhead considerations (time and space) also come into play here, which could make log generation even slower than it is now.  I've never really looked at that enumeration code so perhaps there are some optimization opportunities that could make that path faster and more efficient.

Nevertheless, we'll have to sweep through the tracing code to identify all of the places where an enumerated name might go.  Some refactoring at each trace message might be required.

My initial (and still current) plan is to sweep through the code and simply make whatever we're generating now consistent (more or less a search and replace).  I'll be opening an initial PR soon to demonstrate a couple of ideas and get people's opinions on how it looks before I carry it too far, but there's nothing radical there.  However, since I'm going through the code anyway, rather than using a generic function pointer macro for the format specifier I could create format specifier macros specifically for Nodes, Symbols, Instructions, etc. to act as simple eyecatchers to make them easier to identify for future refactoring work.
